### ISSUE TITLE
Context Menu For Each Folder And Item

### ISSUE THREAD

**Issue by hussain7006** [lukasbach/react-complex-tree#349]
Can you please add the OnContextMenu feature in the tree for each folder and item

**Comment by lukasbach** [comment]
Can you elaborate on what you suggest as a feature? react-complex-tree already allows customizing the render logic of tree items, you can just register your own context menu handlers there.

https://rct.lukasbach.com/docs/guides/rendering

**Comment by hussain7006** [comment]
Yes, I created the context menu but not able to add the folder or items in
a tree using context menu. Means context menu having the option create
folder and delete folder etc.
Thanks

**Comment by lukasbach** [comment]
As mentioned, you can customize the render logic and add custom logic to your rendered elements like context menus from there. If you want changes to the tree structure to be possible, you need to manage the state of the tree externally, and let RCT know of any changes done through change events. There are multiple ways for how to do this, this page documents how to do this for static tree data providers for example: https://rct.lukasbach.com/docs/guides/static-data-provider
